### PR TITLE
Clean up game_modes.c and game_states.c

### PR DESCRIPTION
**Pull request by JoshDuMan**

### CODE REVIEW

File: src/game_states.c [hunk 1]
```diff
@@ -1,146 +1,154 @@
...
+// Unused
+void set_game_mode_flag_4(s32 i) {
+    gMainGameMode[i].flags |= MODE_FLAG_4;
 }
 
-void func_80112E24(s32 i) {
-    gMainGameState[i].flags &= ~0x1C;
+// Unused
+void clear_game_mode_unk_flags(s32 i) {
+    gMainGameMode[i].flags &= ~(MODE_FLAG_3|MODE_FLAG_4|MODE_FLAG_5);
```
Comment (HailSanta) [inline src/game_states.c:103]: spaces around |

File: src/game_modes.c [hunk 2]
```diff
@@ -25,82 +23,237 @@ s32 D_80077420[] = {
...
+    (GameMode*) &GameModeIntro,
+    (GameMode*) &GameModeDemo,
     NULL,
     NULL,
 };
 
-SHIFT_BSS s16 gameMode;
-
-GameMode* set_game_mode_slot(s32 i, GameMode* mode);
+SHIFT_BSS s16 activeGameMode;
```
Comment (HailSanta) [inline src/game_modes.c:250]: CurGameMode would be more consistent with the rest of the codebase

File: src/game_modes.c [hunk 3]
```diff
@@ -25,82 +23,237 @@ s32 D_80077420[] = {
...
+    state_init_demo,
+    state_step_demo,
+    NULL,
+    state_drawUI_demo,
+    #ifdef AVOID_UB
+    NULL,
+    #endif
+};
 
 GameMode* gameModeMap[] = {
```
Comment (HailSanta) [inline src/game_modes.c:227]: See my comment in discord re: explicit indices and omitting the extra NULLs

File: src/game_modes.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+#define _GAME_MODES_H_
+
+#include "common.h"
+
+// These flags are mostly unused. Not enough context to discern their original purpose.
+enum GameModeFlags {
+    MODE_FLAG_NONE           = 0,
+    MODE_FLAG_INITIALIZED    = (1 << 0), // Set when the mod is initialized
+    MODE_FLAG_STEP_NOT_DONE  = (1 << 1), // Turned off after the first step is done
+    MODE_FLAG_3              = (1 << 2),
```
Comment (HailSanta) [inline src/game_modes.h:11]: We typically express unk flags by their bit values in hex, so these 3/4/5 would be 04/08/10

**Review by ethteck — APPROVED** [review]

**Pull request merged**